added dir attribute entry

RobinLilly opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description
Added new Terms directory within content/html/concepts/attributes as well as an entry for the dir attribute.

### Type of Change

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x ] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+---
```
You need to delete all the extra line breaks in the metadata.

RobinLilly replied:
complete

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
````diff
@@ -0,0 +1,64 @@
+<element  dir="ltr | rtl | auto">
+```
+
+## Possible Attribute Values:
````
```suggestion
The `dir` attribute has the following possible values:

```

RobinLilly replied:
When I submitted the original pull request, you suggested I write it this way. I think it looks better too. But if you still prefer, Ill change it

SSwiniarski replied:
@RobinLilly Let's do it this way.

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+
+## Example:
+
+\<p dir="rtl">This paragraph is in English but incorrectly goes right to left.\</p>
```
The example needs to be put into a ```html block.

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+
+\<p dir="rtl">This paragraph is in English but incorrectly goes right to left.\</p>
+
+#### Output:
```
```suggestion
This results in the following output:
```

RobinLilly replied:
complete

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+
+## Example:
+
+\<p dir="ltr">This paragraph is in English and correctly goes left to right.\</p>
```
Again, this needs to be in a ```html block.

RobinLilly replied:
complete (I think)

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+
+\<p dir="ltr">This paragraph is in English and correctly goes left to right.\</p>
+
+#### Output:
```
```suggestion
This results in the following output:
```

RobinLilly replied:
complete

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+
+<p  dir="rtl">This paragraph is in English but incorrectly goes right to left.</p>
+
+## Example:
```
```suggestion
## Example 2:
```

RobinLilly replied:
complete

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
```diff
@@ -0,0 +1,64 @@
+
+>  **Note:** While declaring the directionality of the document in the html tag, don't forget to also declare the language of the document using the `lang` attribute. However, do not make the mistake of assuming that language declarations indicate directionality, or vice versa! Even if the language declaration has a script tag it won't affect the directionality of the text in the user agent. The  ...
+
+## Example:
```
```suggestion
## Example 1:
```

RobinLilly replied:
complete

SSwiniarski reviewed (changes requested):
@RobinLilly I did a review and left a few comments.

RobinLilly commented:
So do I need to make another pull request or do you just look at my latest 2 new commits?

SSwiniarski commented on content/html/concepts/attributes/terms/dir/dir.md:
````diff
@@ -0,0 +1,51 @@
+<element  dir="ltr | rtl | auto">
+```
+
+## Possible Attribute Values: 
````
Can we remove this header and replace it with a sentence like, "These are the possible attribute values:"

RobinLilly replied:
@SSwiniarski I'm sorry I misunderstood which way you wanted it written.
Just to clarify, I only need to make one initial PR to any remote repo, and subsequently only pushes will be necessary , once I pull down any updates?
Did I finally get the project completed?

SSwiniarski replied:
Yes you only need the one PR for this entry. If you decide to add _another_ entry, then you create a new branch of your main fork and do another PR.

SSwiniarski reviewed (changes requested):
@RobinLilly you do not need to submit a new PR. However, please review my one remaining comment.

SSwiniarski reviewed (approved):
@RobinLilly Looks good for a second review.

RobinLilly commented:
@SSwiniarski So do I just wait for another reviewer?

SSwiniarski commented:
@RobinLilly that's right. Another maintainer will come by for a second review. (Every PR has at least two, except for the simplest typo fixes.)

RobinLilly commented:
@yangc95 My pull request is ready for a second review. Would you take a look at it please?

RobinLilly commented:
@KTom101 I have completed this entry and I need another review to be completed. Would you take a look please?

KTom101 commented:
@RobinLilly Sure, I'll let you know when I'm done!

KTom101 reviewed (changes requested):
@RobinLilly I've added a second review. There is mainly restructuring and also rephrasing of two sections I've flagged (2nd paragraph & note in syntax) that seemed to be verbatim of what was said in this W3C article https://www.w3.org/International/questions/qa-html-dir 

Codecademy has a zero-tolerance policy for plagiarism and the use of text from another source is prohibited. These sections must be changed before moving forward.

cc @SSwiniarski

RobinLilly commented:
@KTom101
After my last git push, I have this error: 
Error in user YAML: (<unknown>): did not find expected key while parsing a block mapping at line 1 column 1

I know I can backtrack in my local repo to fix the code but how do I backtrack what I have already pushed to Github?

Do I backtrack locally and then just push again?

KTom101 commented:
@RobinLilly It was throwing the error because of the apostrophe in "element's" within the description. It was seeing it as the end of the description since the description is surrounded by single quotes. Rather than switching to double quotes I just changed the wording so there is no longer an error.

KTom101 commented on content/html/concepts/attributes/terms/dir/dir.md:
````diff
@@ -0,0 +1,57 @@
+```html
+<p dir="rtl">This paragraph is in English but incorrectly goes right to left.</p>
+```
+This results in the following output:
````
```suggestion

This results in the following output:
```

KTom101 commented on content/html/concepts/attributes/terms/dir/dir.md:
````diff
@@ -0,0 +1,57 @@
+```html
+<p dir="ltr">This paragraph is in English and correctly goes left to right.</p>
+```
+This results in the following output:
````
```suggestion

This results in the following output:
```

KTom101 reviewed (approved):
@RobinLilly There are a couple of spots that need line breaks but everything looks good so far. After you add those, will you be able to [format and lint](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#:~:text=The%20verify_formatting%20is%20failing) the file so the checks pass?

KTom101 commented:
@RobinLilly Would you like me to run the formatting and lining for you?

RobinLilly commented:
> @RobinLilly Would you like me to run the formatting and lining for you?

If this is something I should always do, then I definitely want to learn how to do it myself. HOWEVER, I need help with it. I don't remember being taught this in Codecademy.

This is from Contributing.md:

How can I test my changes before submitting a PR?
There are a few tools you'll need to install in order to run our test suites locally:

- [yarn](https://classic.yarnpkg.com/lang/en/docs/install): we use yarn to manage dependencies (instead of npm)
       **I enabled corepack and installed yarn through the command line.**

- [node 16.0.0 (or newer)](https://nodejs.org/en/download/): the tests will likely run on earlier versions of node, but if you'd like your changes to be formatted automatically when you commit then you will need to be on node 16 or greater
      **I installed node.js**

- [nvm](https://github.com/nvm-sh/nvm#installing-and-updating): while not required, we recommend using nvm to manage multiple versions of node on the same machine
     **Do I need this?**

With everything installed, in the repository root (whatever directory contains the package.json file) you can run
     **I'm lost from here**

Note: If you use npm instead of yarn, it is fine to do so with the commands described below.

yarn format:verify and yarn lint:md to validate markdown file contents
yarn test to validate content/ directory structure

KTom101 commented:
nvm is recommended and is used to allow different versions of node. This is useful in cases where certain features are only available in older versions that are not available in the newer versions, or vice versa. Depending on your project, there might be a feature you need or your code might break.

For automatic formatting, you need node 16.0.0 or newer and yarn installed. Every time you commit a change from the command line it should automatically run. You should see something like this: 

<img width="424" alt="Screen Shot 2023-03-01 at 4 11 16 PM" src="https://user-images.githubusercontent.com/40783324/222269910-00d1898f-8d86-4df5-b49b-0d2a698db581.png">

Then you can push the changes up. 

If you don’t do automatic formatting, then you can run the formatting manually yourself by running these commands:

`npm i`
`npm run format:verify`
`npm run format:all`

Then add, commit, and push the changes up.

yangc95 commented on content/html/concepts/attributes/terms/dir/dir.md:
````diff
@@ -0,0 +1,51 @@
+```
+This results in the following output:
+
+<p  dir="ltr">This paragraph is in English and correctly goes left to right.</p>
````
@RobinLilly Please see this comment again

yangc95 commented on content/html/concepts/attributes/terms/dir/dir.md:
````diff
@@ -0,0 +1,51 @@
+```
+This results in the following output:
+
+<p  dir="rtl">This paragraph is in English but incorrectly goes right to left.</p>
````
@RobinLilly Please revisit this comment

yangc95 commented:
I made a quick edit [here](https://github.com/Codecademy/docs/pull/1600/commits/9832935fded85fbe691213ebbde3e6df9112c265). However, please revisit two of @KTom101's comments about the images for both examples 🙂 Please let us know if you have any questions. You can refer to the section on [file hosting](https://github.com/Codecademy/docs/pull/1600/commits/9832935fded85fbe691213ebbde3e6df9112c265) inside the Content Standards.

RobinLilly commented:
@KTom101 
I keep getting this when I try the command npm i

$ npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: codecademy-docs@1.0.0
npm ERR! Found: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.25.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^6.5.1" from @codecademy/eslint-config@3.1.7
npm ERR! node_modules/@codecademy/eslint-config
npm ERR!   dev @codecademy/eslint-config@"^3.1.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\Robin\AppData\Local\npm-cache\_logs\2023-03-03T18_03_10_984Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Robin\AppData\Local\npm-cache\_logs\2023-03-03T18_03_10_984Z-debug-0.log

KTom101 commented:
@RobinLilly It looks like the dependency versions may be off. Can you try `npm install --force` instead?

Also, after you run `npm run format:verify` and `npm run format:all` you can run `git status` to make sure the file has been modified first before adding the file and making a commit.

KTom101 commented:
It looks like you activated the husky automation test tools in the process. There is a section in the middle that says:

```
codecademy-docs@1.0.0 prepare husky install
husky - Git hooks installed
added 683 packages, and audited 684 packages in 32s 
```

Can we check if it indeed installed, first? To check this, you would have to commit something since that’s when the tests are activated. 

Add an extra space at the end of a line somewhere so that the file is shown as modified when `git status` is ran and then add and commit. 

If you see something like this:

![image](https://user-images.githubusercontent.com/40783324/222811609-3e1806ea-207a-4c97-91a2-de9132c8fe16.png) 

then the file has automatically been formatted for you. If not, we can try fixing the dependency tree error again.

RobinLilly commented:
This is what I got. I have not pushed yet

Robin@DESKTOP-JUNB5CU MINGW64 ~/Desktop/coding-projects/Git-Practice/robins-local (robins-branch)
$ git commit -m "add space after punctuation note"
[STARTED] Preparing...
[SUCCESS] Preparing...
[STARTED] Running tasks...
[STARTED] Running tasks for *.md
[STARTED] npm run format
[SUCCESS] npm run format
[STARTED] markdownlint
[SUCCESS] markdownlint
[SUCCESS] Running tasks for *.md
[SUCCESS] Running tasks...
[STARTED] Applying modifications...
[SUCCESS] Applying modifications...
[STARTED] Cleaning up...
[SUCCESS] Cleaning up...
[robins-branch 3bf243f3] add space after punctuation note

KTom101 commented:
@RobinLilly Ok, it does look like it's working. Try pushing it, please.

RobinLilly commented:
@KTom101 
I pushed. How do I know if it passed the tests?

$ git push origin robins-branch
Enumerating objects: 13, done.
Counting objects: 100% (11/11), done.
Delta compression using up to 8 threads
Compressing objects: 100% (7/7), done.
Writing objects: 100% (7/7), 24.34 KiB | 3.48 MiB/s, done.
Total 7 (delta 5), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (5/5), completed with 4 local objects.
To https://github.com/RobinLilly/codecademy-docs.git
   642ca1b1..8cc4c77b  robins-branch -> robins-branch

RobinLilly commented:
@KTom101 
I'm so sorry I'm being such a pain. 

$ git status
On branch robins-branch
Your branch is up to date with 'origin/robins-branch'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        deleted:    package.json
        deleted:    yarn.lock


Robin@DESKTOP-JUNB5CU MINGW64 ~/Desktop/coding-projects/Git-Practice/robins-local (robins-branch)
$ git commit -m "removed files"
Configuration could not be found.
husky - pre-commit hook exited with code 1 (error)

KTom101 commented:
@RobinLilly It's a little hard to tell exactly what you're doing, but I think you might be removing it completely from your repo (hence the error code on the last line). It just needs to be removed from the PR, but the files should be kept on your local repo. For example, this is how my file structure looks locally:

<img width="751" alt="Screenshot 2023-03-07 at 1 01 06 AM" src="https://user-images.githubusercontent.com/40783324/223337873-8a122cdd-d684-4e01-8747-5dfa5c2b1f41.png">

The `yarn.lock` and `package.json` files are there.

You may have to [reset](https://www.codecademy.com/resources/docs/git/reset) this commit to get back to where you were and then remove the file from the PR following [these steps](https://stackoverflow.com/questions/39459467/remove-a-modified-file-from-pull-request#:~:text=Switch%20to%20the%20branch%20from%20which%20you%20created%20the%20pull%20request%3A).

RobinLilly commented:
What does it look like now? 
Is it ok to run the git commands from VSCode's terminal?
If this one doesn't work, should I just move the file to a different location, delete my local and remote repos, and re-fork and re-clone and start over from scratch. I probably made a mistake when I was having trouble installing node.js and yarn.

KTom101 commented:
@RobinLilly You would lose this branch in the process of doing that. Perhaps, I can just take this over from my end and then you can try again on your next entry?

RobinLilly commented:
I really want to finish the task. What would be the downside of losing the branch? Is it just the commit history? I'll still have the markdown file saved in a different location. I understand though if you think it would be more trouble than it's worth or if my idea of scrapping it all would be frowned upon.

KTom101 commented:
@RobinLilly You would have to create a new pull request if you re-fork and re-clone because the branch will be lost. I don't feel for this case it's necessary because the only issue is the `yarn.lock` and `package.json` files just need to be removed from only the PR. 

You will end up in the same situation after re-forking and re-cloning because the `yarn.lock` file was added and the `package.json` file was modified after installing `npm`. Then the files will need to be removed before pushing the changes using `git checkout -- file-name`

RobinLilly commented:
@KTom101 Ok just fix it for me please.
I'll look for another entry to work on. 
Thank you so much for all your help. 
Sorry I needed so much hand-holding.
This was my first time.

KTom101 commented:
@RobinLilly No worries!  Everything looks good now. In commit [0045015](https://github.com/Codecademy/docs/pull/1600/commits/0045015697c3e5e1911295ef548034a3730d2a3b), I removed the files from the PR with the commands `git checkout origin/main -- yarn.lock` and `git checkout origin/main -- package.json `. Once the files were removed, all the checks passed. This confirms you do have the automatic formatting up and running and the checks were fixed, the files just needed to be removed to reflect this.

I also made an edit to the links in commit [813f565](https://github.com/Codecademy/docs/pull/1600/commits/813f5651b718d011c17e7f12fbbc8aa329832bb6)so that they start with "https://raw.githubusercontent.com/Codecademy/docs/main/media/" which is the correct format for links (mentioned in the style guide [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#file-hosting:~:text=Images%20should%20be%20linked%20to%20using%20the%20following%20url%3A%20%22https%3A//raw.githubusercontent.com/Codecademy/docs/main/media/%22%20followed%20by%20the%20image%20name.))

KTom101 commented:
@RobinLilly Made one more edit to the tags in commit [21493c4](https://github.com/Codecademy/docs/pull/1600/commits/21493c4eabb153a1c2f47a87a7da0e9e1476c0f8). Entry is now live here: https://www.codecademy.com/resources/docs/html/attributes/dir You're all set and you're listed as a contributor! 🎉

<img width="850" alt="Screenshot 2023-03-09 at 2 38 54 AM" src="https://user-images.githubusercontent.com/40783324/223952990-93a4650c-b58c-49d5-93bc-e843f0e19c9f.png">

Thank you for contributing and I'll be looking out for your next entry! 

cc @yangc95 @SSwiniarski
